com.unity.ml-agents/Runtime/Sensors/GridSensorComponent.cs

```diff
@@ -213,29 +213,33 @@ public override ISensor[] CreateSensors()
             m_DebugSensor = new GridSensorBase("DebugGridSensor", m_CellScale, m_GridSize, m_DetectableTags, SensorCompressionType.None);
             m_BoxOverlapChecker.RegisterDebugSensor(m_DebugSensor);
 
-            var gridSensors = GetGridSensors();
-            if (gridSensors == null || gridSensors.Length < 1)
+            m_Sensors = GetGridSensors().ToList();
+            if (m_Sensors == null || m_Sensors.Count < 1)
             {
                 throw new UnityAgentsException("GridSensorComponent received no sensors. Specify at least one observation type (OneHot/Counting) to use grid sensors." +
                     "If you're overriding GridSensorComponent.GetGridSensors(), return at least one grid sensor.");
             }
 
-            foreach (var sensor in gridSensors)
+            // Only one sensor needs to reference the boxOverlapChecker, so that it gets updated exactly once
+            m_Sensors[0].m_BoxOverlapChecker = m_BoxOverlapChecker;
```

Comment:
Is this correct?  Won't only one sensor have anything to write?  Will it actually be a stacked observation?